tests/script_annotations_outputs/script_annotations_output_param_and_artifact_in_func.py

```diff
@@ -0,0 +1,24 @@
+try:
+    from typing import Annotated  # type: ignore
+except ImportError:
+    from typing_extensions import Annotated  # type: ignore
+
+from pathlib import Path
+
+from hera.workflows import Artifact, Parameter, Workflow, script
+from hera.workflows.steps import Steps
+
+
+@script()
+def script_param_in_fun_sig(
+    a_number: Annotated[int, Parameter(name="a_number")],
+    successor: Annotated[Path, Parameter(name="successor", output=True)],
+    successor2: Annotated[Path, Artifact(name="successor2", output=True)],
```

Comment:
From a UX perspective it might be beneficial to have the `Parameter.name` field, and `Artifact.name`, inferred based on the arg. Is that possible? Even if it means setting an arbitrary name on init then changing it, perhaps?

Reply:
```python
    a_number: Annotated[int, Parameter()],
    successor: Annotated[Path, Parameter(output=True)],
    successor2: Annotated[Path, Artifact(output=True)],
```
looks a bit nicer!